src/ng/directive/input.js

```diff
@@ -1087,6 +1116,24 @@ function textInputType(scope, element, attr, ctrl, $sniffer, $browser) {
   stringBasedInputType(ctrl);
 }
 
+function createKeyupListener(type, callback) {
+  // A list containing single keyCodes and keyCode-ranges (in the form [min, max])
+  var keys = KEYS_PER_DATE_INPUT_TYPE[type];
+
+  return function keyupListener(evt) {
+    // Ignore if a modifier key is also pressed
+    if (!evt || evt.altKey || evt.ctrlKey || evt.metaKey || evt.shiftKey) return;
+
+    var key = evt.keyCode;
+    var affectsInput = keys.some(function(keyOrRange) {
+      return !isArray(keyOrRange)
+          ? (key === keyOrRange) : (keyOrRange[0] <= key) && (key <= keyOrRange[1]);
+    });
+
+    if (affectsInput) callback('input');
```

Comment:
Does passing `'input'` here do anything? It seems like this is only used doing `event = ev && ev.type` which then gets passed as the "trigger" to `$setViewValue`, so it will be undefined in this case...